tattle/engine.py

```diff
@@ -18,45 +18,165 @@
 import time
 import yaml
 import sys
+import getpass
 
 from tattle.model import QueryConfig
 from tattle.model import Query
 from tattle.model import Filter
 
-
 GITHUB_USER = 'GITHUB_USER'
 GITHUB_PASS = 'GITHUB_PASS'
 CONFIG_PATH_COMMAND_NAME = '--config-path'
 CONFIG_PATH_HELP_TEXT = 'a path to a YAML configuration file'
-
+ORG_NAME_COMMAND_NAME = '--org'
+ORG_NAME_HELP_TEXT = 'the organization to be searched upon'
+FILTERS_COMMAND_NAME = '--filters'
+FILTERS_HELP_TEXT = 'filters to be used. The input should be enclosed by ' \
+                    '[ ] brackets. All filters will be separated by a comma.' \
+                    ' Each filters are enclosed in brackets. For example: ' \
+                    'tattle --filters ' \
+                    '\"[{\'name\': [\'CFY\']}]\"'
+BRANCH_NAME_COMMAND_NAME = '--branch-names'
+NAME_HELP_TEXT = 'the names to be searched in an intersection'
+JIRA_TEAM_COMMAND_NAME = '--jira-team'
+JIRA_TEAM_HELP_TEXT = 'the jira team to be searched'
+JIRA_STATUSES_COMMAND_NAME = '--jira-statuses'
+JIRA_STATUSES_HELP_TEXT = 'a list of all the JIRA status names that won\'t' \
+                          ' be filtered by that issue filter, separated ' \
+                          'with space. If we refer to the above example, ' \
+                          '"--jira-statuses Closed Resolved"' \
+                          ' will filter all branches whose corresponding ' \
+                          'JIRA issue is not ' \
+                          '\'Closed\' or \'Resolved\''
 GITHUB_ENV_VARS_DONT_EXIST = 'GitHub authentication environment variables' \
                              ' do not exist.\nPlease define them: ' \
                              '[GITHUB_USER], [GITHUB_PASS], and try again.'
+OUTPUT_PATH_COMMAND_NAME = '--output-path'
+OUTPUT_PATH_HELP_TEXT = 'the path of tattle\'s product, the report.json file' \
```

Comment:
the path of tattle\'s output file, a report.json file. If not specified, /tmp/tattle/report.json will be used.